server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -115,20 +115,34 @@ protected boolean shouldRefreshFileState(ClusterState clusterState) {
      */
     @Override
     protected void processFileChanges() throws ExecutionException, InterruptedException, IOException {
-        PlainActionFuture<Void> completion = new PlainActionFuture<>();
         logger.info("processing path [{}] for [{}]", watchedFile(), NAMESPACE);
+        processFileChanges(false);
+    }
+
+    /**
+     * Read settings and pass them to {@link ReservedClusterStateService} for application.
+     * Settings will be reprocessed even if the cluster-state version equals that found in the settings file.
+     */
+    @Override
+    protected void processFileOnServiceStart() throws IOException, ExecutionException, InterruptedException {
+        logger.info("processing path [{}] for [{}] on service start", watchedFile(), NAMESPACE);
+        processFileChanges(true);
+    }
+
+    private void processFileChanges(boolean allowSameVersion) throws IOException, InterruptedException, ExecutionException {
```

Comment:
nit: I think `forceReprocess` would be a good name since we already "allow" the same version, we just don't do anything if it's the same. Alternatively `reprocessSameVersion`.

Reply:
yup, agreed `reprocessSameVersion` reads better 👍